docs/source/get_involved/code_of_conduct.rst

```diff
@@ -28,40 +29,59 @@ conform to the following Code of Conduct.
 1 Introduction
 --------------
 
-sktime is a community-oriented and -led project. We value the
-involvement of everyone in the community. We are committed to creating a
-friendly and respectful place for learning, teaching and contributing.
-All participants in our in-person events and online communications are
-expected to show respect and courtesy to others at all times.
+sktime is a community-oriented and -led project.
+
+We value the involvement of everyone in the community, and strive to be an open
+community. We commit to provide fair and revarding opportunities for everyone without
+restriction, to participate, learn, and become leaders of the community. We expect from
+members of our community to create such opportunities for others as have been created
+for them.
```

Comment:
interesting question. I would say: it should be an expectation, according to engagement and "power". I.e., a new contributor or junior career person is not expected to since it is not in their power to. A very senior academic or industry data scientist who wants to have "power" in `sktime` (e.g., on the CC, or core developer) is expected to mentor, donate, or create opportunities otherwise, in transparent and non-exclusionary manner.

Reply:
I tend to agree with what Lovkush is suggesting here though.  I agree it should be a community standard/supported and value in the community, but it doesn't quite feel like a code of conduct thing? Or something that could generate a Code of Conduct violation.  I think doing so leads to a situation that can be hard to equitably enforce without structured oversight/could run the risk of undermining other more "traditional" sorts of complaints people might raise as CoC violations?

Perhaps there is another way to communicate this though? ie spelled out more clearly in terms of what you sign up for by becoming a core dev or CC member?

Reply:
> ie spelled out more clearly in terms of what you sign up for by becoming a core dev or CC member?

hm, yes, I think that might be an option

> I agree it should be a community standard/supported and value in the community, but it doesn't quite feel like a code of conduct thing? 

Hm, I *do* think it is a code of conduct thing. The situation I have in mind is the "negative" of it, where someone hijacks community assets for their personal network, actively prevents opportunities out of turf mentality, keeps certain people "out" due to base motivation (e.g., chauvinism).

Systemic versions of racism and sexism nowadays don't take the form of communication (which the old CoC focuses on mostly), but denial of resource, denial of opportunities, shrouded in plausible deniability.

The systemic versions appear when we talk about substantial resource allocation decisions, in the order that motivates this CoC change.

Example of racism, "old CoC type": open name calling in a social media thread; talking over people in a meeting. That is very communication focused.

Example of racism, "new CoC type": advertising a position in a way that people outside socially connected circles, with certain nationalities (outside affluent countries) or outside non-diverse "ingroup networks" will have a substantially harder time to find the opportunity, apply for it, or succeed, assuming equal level of competence.

Plausible deniability: "oh, admin says we have to do X" or "I did advertise"

Silent exclusionary measures: tailoring the job description in a very specific way, advertising only in low visibility outlets, deferring exclusion to the set-up of the position (e.g., causing problematic visa requirements), restricting timelines to make applicaion outside direct network difficult.

None of this is caught well by the "old style CoC", because all communication was extremely polite.

Reply:
Important: the motivation of systemic exclusion is typically hard to spot and near impossible to prove.

*but*, the exclusionary measures are easy to spot and can be put in a code of conduct as forbidden. Hence the discussion of paper trail, transparency, etc.

Reply:
In fact, I think code of conduct may be the only tool we have to regulate intentional or accidental of systemic exclusion as discussed above.

Reply:
Basically I still think if it is going to be something that you want to be enforcable then it needs to be a bit more specific (and people need to opt in) otherwise I still think it should be something that is under community culture/guidelines and/or core dev agreements.  I very much agree that "lack of opportunities" and other related issues you bring up here are very important and certainly can impact someone's experience with `sktime` and software development more broadly, however I don't necessarily think that is the point here.  To me the question is - what is the best way to encourage this within the community? 

I don't believe that putting it in the Code of Conduct and making it a potentially enforceable offence is the way to go.  Here are my reasons why:
- As a mostly volunteer organization people will inevitably be involved in sktime to various degrees, how do you quantify how much each individual should be giving to the community/creating opportunities?  Ultimately unless you are specific there will be conflicts, because a persons assessment of their own skill, commitment amount, and seniority will not perfectly line up with anyone else's. On top of that it can be tricky to compare or measure how much someone is contributing to the community/creating opportunities...
- It stands apart from the other expectations in the Code of Conduct because rather than asking people *not* to do something it is asking people *to* do something.  Which again I think is hard to expect/enforce in a volunteer community. Maybe it is something that you can ask of core devs or people on CC/CoC, but I think it should be framed as an opt-in responsibility rather than a general Code of Conduct thing. 
- Lastly, because of the above vague nature of expectations/results/whether or not people have opted in I do worry that it could be used (or be seen to be used) unfairly.  I do think this runs the risk of alienating certain individuals if they are not on board with this change, and I think it runs the risk of making the CoC as a whole be seen as a document that can be used for political issues, especially if it boils down to situations like you allude to above of plausible deniability vs CoC....  I am not saying that is the intent, I just can imagine it being seen that way, and I think it could undermine the underlying message/goal.

However, if you were able to make some of these expectations more concrete then I could be in favor.  For example:
 - if you are creating an opportunity that makes use of `sktime` resources then that opportunity should be shared and advertised with all of the sktime community. 
 - if you are open to mentoring individuals within the `sktime` community then you should open that opportunity up for anyone who wants to engage/apply!  
 - etc etc

Reply:
> It stands apart from the other expectations in the Code of Conduct because rather than asking people not to do something it is asking people to do something

Hm, is that really so? The CoC has both positives and negatives already. Entire section "expected behaviour" - "be respectful", "use inclusive language", etc, these are positive asks "to do sth" in my reading?

> because of the above vague nature of expectations/results/whether or not people have opted in I do worry that it could be used (or be seen to be used) unfairly.

Of course - every expectation needs to be as specific as possible, or can be used as "rubber conditions" arbitrarily. The legal principle here is [legal certainty](https://en.wikipedia.org/wiki/Legal_certainty).

We need something unambiguous, I think the negative examples in section 2.2 are pretty good in being concrete.

Question: what is so different between

"written or verbal comments which have the effect of excluding people on the basis of protected characteristics"

and 

"decisions on resource allocation or operations which have the effect of excluding people on the basis of protected characteristics"

(note: in the current condition focused on communication, intention is also not a condition for it to apply!)

> Lastly, because of the above vague nature of expectations/results/whether or not people have opted in I do worry that it could be used (or be seen to be used) unfairly. I do think this runs the risk of alienating certain individuals if they are not on board with this change

I think there are two issues in this sentence:

* worry that the rules will be used unfairly
* worry whether everyone is onboard with the rules

These are two things, we should separate them!

"used unfairly": fairness must be built in the design, I agree. We can borrow principles from legal theory. Number one is legal certainty (see above), conditions must not be vague. Number two is also important imo - separation of powers. The CoC committee that judicates should be separate from the CC, and, optimally, unconnected to any network.
This will not prevent political use (nor does it in "real life"), but the political influence in judication is minimized.

"everyone is onboard": there is a chicken/egg problem here. The document is the very definition of what the community considers as "good" or "fair". Formally, the CC can decide this, not the entire community needs to be on board.

This might sound harsh, but let me explain by example my stance on this.
In a hypothetical situation, there surely exist, somewhere, people who like making racist or sexist jokes a lot and consider that kind of environment as positive ("boys club atmosphere"). I have been told one doesn't need to look to far in tech...

Personally, I would not shead a single tear, a person like that can leave. My vision of `sktime` is one of a cohesive, but professional, respectful, and diverse community.
Same thing for people who hire only friends and disregard merit or fair opportunities. Or people who can't be convinced to commit to even the most elementary anti-corruption measures and transparency requirements. In a sense that's just the systemic version of the "boys club".
One can't help that standing up against that will inevitably be framed as "political" or "toxic" or similar. It's also sad times we live in that an anti-corrupt stance is not broad consensus, but considered "political".

Reply:
> and transparency requirements

Several months ago, I put suggestion forward that all core devs ought to reveal any affiliations and how their efforts are paid for (e.g. voluntary, part of phd, part of job, etc.).  This is to help community know if there are any major outside influences and to prevent any 'takeovers'.  I think only Franz and I liked the suggestion (despite the impetus for this being that both Franz and I (as well as other contributors) were at Shell, so I thought it was useful for community to be aware of this).

I dont think anybody explained why they disliked it. But I never pushed for an explanation either.

Sad indeed that striving for best practice is 'political'! But that is the way of the world.

Reply:
I do agree that whatever you and the CC committee decide can/should be incorporated into this document. :)

In truth I kind of thought/hoped this was more a discussion about the *best* way to encourage people to make more opportunities within `sktime` and make the community more open/diverse!  Not sure why it felt like we kept having a discussion about whether or not "boys club's" or sexist jokes should be condoned...

But at the end of the day I really think we are on the same page with goals and if leadership think this is the best way to go then I am all for it! :)  In truth this is probably a small point and the discussion over it has probably grown out of proportion. 

Interesting that the disclosure of ties didn't go over well?  Maybe it could be something worth bringing up again in the future.

Reply:
numfocus guidance is to (a) disclose affiliation for CC members, and (b) contractual stipulation that no two members can be from the same group/entity.

We could make it a requirement for CC members, but I suppose that's a separate discussion.

I have added now a section on best practice and transparency.